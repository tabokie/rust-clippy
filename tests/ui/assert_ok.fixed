// run-rustfix
#![warn(clippy::assert_ok)]

fn main() {
    // basic case
    let r: std::result::Result<(), ()> = Ok(());
    r.unwrap();
}
