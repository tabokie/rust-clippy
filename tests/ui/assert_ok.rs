// run-rustfix
#![warn(clippy::assert_ok)]

fn main() {
    // basic case
    let r: std::result::Result<(), ()> = Ok(());
    assert!(r.is_ok());
}
