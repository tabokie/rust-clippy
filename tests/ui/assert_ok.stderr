error: `assert!(r.is_ok())` gives bad error message
  --> $DIR/assert_ok.rs:7:5
   |
LL |     assert!(r.is_ok());
   |     ^^^^^^^^^^^^^^^^^^ help: replace with: `r.unwrap()`
   |
   = note: `-D clippy::assert-ok` implied by `-D warnings`

error: aborting due to previous error

